Updates to Coolpoints UI elements

KamalJDavis opened the pull request:
- [x]  Icon svg file linked in layout.pug and acp.pug.

- [x] redesigned coolpoints button and reimplimented logic on page load.

- [x]  New animations for gaining and losing CP.

- [x]  Removed gold-bar.gif for goldbar.avif 

- [x]  Restructured Coolpoints counter and user button (combined)

- [x] Started repairing the Coolpoints modal.

KamalJDavis committed:
+ Icon svg file linked in layout.pug and acp.pug.
+ redesigned coolpoints button and reimplimented logic on page load.
+ New animations for gaining and losing CP.
+ Started repairing the coolpoints page.
- Removed gold-bar.gif for goldbar.avif
o Restructured Coolpoints counter and user button (combined)

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -23,10 +23,21 @@
-  $("#coinWrapper").addClass("cpFadeIn");
 }
 
+/**
+ * Kamal - moved the button loading out of "initPointsForSelf()"" since it is called on page load.
+ * If we wanted to put any other animations on the button for points gained or earned we'd need the previous
+ * animation class removed i.e., "cpFadeIn".
+ */
+(function spawnPointsBtn(){
+  const buttonEl = document.getElementById("coolpointsbtn");
+  buttonEl.classList.add("cpFadeIn");
+  Promise.all(
+    buttonEl.getAnimations({ subtree: true }).map((animation) => animation.finished)
+  ).then(() => buttonEl.classList.remove("cpFadeIn")).catch((error) => console.log(error));
+})();
```
concern: This was in `initPointsForSelf` because a user's points aren't always available when you first login. It's not a great solution to have it just fade in but I think this might be a bit worse since now the points "pop in" on load

KamalJDavis replied:
Ah ok, that makes sense. In that case, we can remove the self calling part and just have `initPointsForSelf` still be the caller. Another idea I came up with is to have a guest version of the button, no points are earned but you can still open the modal and all that. We would just spawn the button without the text and just the coin graphic.

streeeggs replied:
Yea that's not a bad idea. Could give them FOMO

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -72,11 +83,14 @@ function animatePointUpdate(ptEl, msgEl, diff) {
   const isPositive = diff > 0;
   const bounceAnimationName = isPositive ? "cpBounce" : "cpShake";
   const fadeAnimationName = isPositive ? "cpFadeGreen" : "cpFadeRed";
+  const glowAnimationName = isPositive ? "glowGreen" : "glowRed";
   const msgText = isPositive ? `+${diff}` : `${diff}`;
-
+  const btnEl = $("#coolpointsbtn");
+  
   msgEl.text(msgText);
-  updateAnimation(ptEl.attr("id"), bounceAnimationName);
-  updateAnimation(msgEl.attr("id"), fadeAnimationName);
+  updateAnimation(btnEl.attr("id"), [glowAnimationName]);
+  updateAnimation(ptEl.attr("id"), [bounceAnimationName]);
+  updateAnimation(msgEl.attr("id"), [fadeAnimationName]);
```
question: why were these changed to pass in an array instead of a string?

KamalJDavis replied:
I was testing something involving multiple effects, but forgot to switch this part back after reverting those changes, lol.

streeeggs commented on templates/channel.pug:
```diff
@@ -63,7 +63,10 @@ html(lang="en")
               button#emotelistbtn.btn.btn-sm.btn-default: span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-emote-dead")
               button#chatoptsbtn.btn.btn-sm.btn-default: span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-ui-preferences")
-              button#coolpointsbtn.btn.btn-sm.btn-default(onclick="javascript:showCoolPointsUserPrompt()"): span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-ui-coin-stencil") CoolPoints
+              button#coolpointsbtn.btn.btn-sm.btn-default(onclick="javascript:showCoolPointsUserPrompt()"): span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-ui-coin-stencil")
+                svg.ch-icon.ch-button-icon.ch-cp: use(xlink:href="#ch-icon-ui-coin")
+                span#coinAmt
```
nit: this could use a little centering IMO
![image](https://github.com/user-attachments/assets/7cfcc982-0730-4e09-ba3f-c4262cb7ccd3)
![image](https://github.com/user-attachments/assets/09037f90-03d3-4ff0-95b5-80395add98aa)
I achieved the second by just making this wrapping span a flexbox and aligning to center

KamalJDavis replied:
Flexbox works most of the time, but I believe the it would begin to wrap below 1024px, which I think is Bootstrap 3's large breakpoint size. The item's centering might be off due to some default Sync/Cytube CSS for line-height. I'll take another look and see if I can pin that down.

streeeggs replied:
Oh yea? I tried it out on a mobile viewport and it looked alright. tried it with 1 CP and 1000000 CP and seemed alright but lmk what you find

streeeggs commented on www/css/coolhole.css:
```diff
@@ -398,17 +398,107 @@
 }
 
 /* CoolPoints User Styling */
-#coinWrapper {
```
nit: While the UI looks _much_ better (thanks for that I have no eye for design), I think there's a few things here that hurt us or could be improved

![image](https://github.com/user-attachments/assets/3f07a034-3791-4e96-bff7-43840143756b)
1. I think we could use some padding here on all the inner content in the modal. The title and the tabs certainly don't need it
2. Might be personal preference but I think I like how the title looks centered
![image](https://github.com/user-attachments/assets/8143cd60-2cd1-44b9-8201-58a6609c11e1)
3. I think the white close icon is good but it's not consistent with the rest of the modals. If we want to change this we might wanna consider finding the style responsible in the theme

![image](https://github.com/user-attachments/assets/895ce27c-4ec5-462e-a558-dd9b810f540d)
Casino real busted (not that it wasn't before)

![image](https://github.com/user-attachments/assets/3a41f12d-dfc3-4c11-adaa-eaaba1b393f4)
Status of actions could use some help if you're willing

KamalJDavis replied:
You read my mind, I'm just fighting all of the Bootstrap 4/ Finalboss, `.row` class abuse, I was already pretty tired last night and tracking down why the modal was padding but the dividers weren't was driving me insane. I do plan to finish centering and revising all this later today.

streeeggs replied:
Looks much better now. Probably worth throwing to others once it's on staging to have 'em poke around; give feedback
![image](https://github.com/user-attachments/assets/eb103b74-0703-4b72-8b66-c0cc80b64aad)
![image](https://github.com/user-attachments/assets/8fe89ad1-680d-4363-b54c-04b87174cd6a)
![image](https://github.com/user-attachments/assets/c13798a7-1f44-4999-935f-7031578b99da)

streeeggs commented on www/css/coolhole.css:
```diff
@@ -398,17 +398,107 @@
 
 /* CoolPoints User Styling */
-#coinWrapper {
+#cpModal {
+  padding: 0;
+}
+#cpModalContent {
+  padding-top: 5px;
+}
+#cpModalBody {
+  padding: 0;
+}
```
nit: we can just use bootstrap's padding classes, no? isn't like `p-0` a thing?

streeeggs replied:
also not a huge fan of the naming since we have multiple `cpModal`s. I'd also prefer classes that describe the style to avoid code repetition unless the style is very specific the element you're providing it against

KamalJDavis replied:
> nit: we can just use bootstrap's padding classes, no? isn't like `p-0` a thing?

Unfortunately, Bootstrap 3 doesn't have those, that's Bootstrap 4 and up

> also not a huge fan of the naming since we have multiple `cpModal`s. I'd also prefer classes that describe the style to avoid code repetition unless the style is very specific the element you're providing it against

I agree with this, mainly wanted to pin down the problems with the CSS and get it laid out properly before reorganizing everything.

streeeggs commented on templates/coolhole-coolpoints.pug:
```diff
@@ -54,37 +54,33 @@ mixin coolpointsusertable
                 +casino
 
 mixin coolpoints-prompt
   #cp-prompt.tab-pane.fade.col-12.active.in(role="tabpanel", aria-labelledby="cpPromptTab")
```
nit: I'd suggest making a common class for each tab content for this prompt. I think some of these general styles like `    .col(style="justify-content: space-evenly; align-items: center;")` could probably be wrapped up in that

KamalJDavis replied:
Github hid this one for some reason...
Anyways, I agree and now I'm thinking of just restructuring the modal to match the other modals for easier CSS changes.

streeeggs commented on templates/layout.pug:
```diff
@@ -5,6 +5,7 @@ html(lang="en")
       include head
       +head()
   body
+    include ../www/svg/coolhole-icons.svg
```
question: does this need to be here? if so, should we remove the include from `channel` then?

KamalJDavis replied:
`channel` and `layout` exist on two separate planes, I forget which `.pug` file does this but only one pug extends specifically from `layout`. Adding it to `head` may have caused some issues, but I can't remember exactly at the moment.

streeeggs replied:
if you find out, lmk. trying to reduce the amount of changes to common files if possible

streeeggs commented on www/css/coolhole.css:
```diff
@@ -398,17 +398,107 @@
+  padding-right: 18px;
+  align-items: center;
+}
+#cpModalHeader > button.close {
+    color: rgb(246, 246, 246);
+    font-size: 24px;
+    opacity: 100%;
+}
```
nit: already said it above but I'd like to see us apply this to the style so it's on every modal, not just this one

streeeggs replied:
also inconsistent tabbing

streeeggs commented on templates/coolhole-coolpoints.pug:
```diff
@@ -140,7 +135,7 @@ mixin under-construction
 mixin legal-nonsense
   span(style="color: gray; font-size: 12px") Terms of Service:
-  div(style="overflow-y: scroll; height: 75px; font-size: 12px; color: gray")
+  div(style="overflow-y: scroll; height: 75px; font-size: 12px; color: gray; background-color: dark-gray;")
```
nit: we could make a style for this. i didn't realize what a mess this inline style was becoming

KamalJDavis replied:
Agreed, we can clinch this in while making the modal CSS more broad.

streeeggs commented on templates/channel.pug:
```diff
@@ -63,7 +63,10 @@ html(lang="en")
               button#emotelistbtn.btn.btn-sm.btn-default: span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-emote-dead")
               button#chatoptsbtn.btn.btn-sm.btn-default: span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-ui-preferences")
-              button#coolpointsbtn.btn.btn-sm.btn-default(onclick="javascript:showCoolPointsUserPrompt()"): span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-ui-coin-stencil") CoolPoints
+              button#coolpointsbtn.btn.btn-sm.btn-default(onclick="javascript:showCoolPointsUserPrompt()"): span: svg.ch-icon(viewbox="0 0 24 24"): use(href="#ch-icon-ui-coin-stencil")
+                svg.ch-icon.ch-button-icon.ch-cp: use(xlink:href="#ch-icon-ui-coin")
```
question: are there two SVGs being rendered here?

KamalJDavis replied:
I think so actually, lol.

streeeggs replied:
i think the old "colored" icon is in there. i removed it and it seemed ok?

KamalJDavis replied:
> i think the old "colored" icon is in there. i removed it and it seemed ok?

Yep it was fine, I didn't even realize it was there since it seemed to be either under the stencil or just pushed somewhere else. I've got an idea to utilize the color icon still without needing an extra element.

streeeggs commented on www/css/coolhole.css:
```diff
@@ -398,17 +398,107 @@
+  height: 50px;
+  margin-right: 10px;
+}
+#coolpointsbtn {
```
nit: inconsistent naming `cpPromptBtn` would be more appropriate

streeeggs reviewed (commented):
I got a bit carried away. A lot of these are nit picks but I think overall the work looks really good. Nice job

streeeggs commented on templates/acp.pug:
```diff
@@ -5,6 +5,7 @@ html(lang="en")
     +head()
     link(rel="stylesheet", type="text/css", href="/css/acp.css")
   body
+    include ../www/svg/coolhole-icons.svg
```
oh one more, does acp need icons? I can't even get ACP to work...
![Uploading image.png…]()

KamalJDavis replied:
That screenshot didn't upload, but I can check that myself in a bit. If it isn't working we can just drop it for now. As I mentioned above, I'll see if placing the `svg` include in head works.

streeeggs replied:
oh annoying. here's the screenshot. and yea heard
![image](https://github.com/user-attachments/assets/1241f1fe-0fab-4d96-8968-d006a80300aa)

KamalJDavis committed:
- Moved SVG include to head, removed from appropriate pug files
- Renamed and refactored css for coolpoints to maintain name schema
- Removed redundancy from SVG file

streeeggs commented on www/js/coolhole-callbacks.js:
```diff
@@ -31,7 +31,11 @@ const CoolholeCallbacks = {
     CHANNEL.usersCoolPoints = response.data;
 
     // Update the UI
+    // Check the users rank and fade in the counter/button
     initPointsForSelf(CLIENT.coolpoints);
+    if (CLIENT.rank >= 1) {
+      initCounterForSelf();
+    }
```
nit: Do we need these to be separated? Also I plan on trying to tackle (more broadly) the issue of non-logged in or ineligible users from getting the `init` callback hit:
https://github.com/streeeggs/coolhole/pull/22/files#diff-9b2696fd14264957ce69042fbba8e415728fb13046d05b9c2d213165526b99d4R271-R272

That being said if we needed these separated and were concerned with a client-side check , I would think it would be a concern both `initPointsForSelf` and `initCounterForSelf`

KamalJDavis replied:
I think that was the main reason I made them separate in the first place. If the user is logged in you get the points counter and button, logged out it is just the button. That way anons and logged out users can still see the modal.
I thought about a button for non logged in users, but not having the counter start seemed a simpler solution.
I can try out some ideas on your branch to see what can be done with the check in place thought.

streeeggs replied:
Yea I see. I would say you should, instead, decouple the button rendering from this callback function. I would suggest adding the "counter" if the user is valid but always try to render the button outside of this callback. Maybe at the top of `coolpoints-utils.js`

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -24,9 +24,19 @@ function initPointsForSelf(pts) {
+function initCounterForSelf(){
+  const buttonEl = document.getElementById("cpPromptBtn");
+  buttonEl.classList.add("cpFadeIn");
+  Promise.all(
+    buttonEl.getAnimations({ subtree: true }).map((animation) => animation.finished)
+  ).then(() => buttonEl.classList.remove("cpFadeIn")).catch((error) => console.log(error));
```
nit: use `updateAnimation` instead

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -37,7 +47,7 @@ function updateAnimation(id, animationName) {
   Promise.all(
     el.getAnimations({ subtree: true }).map((animation) => animation.finished)
-  ).then(() => el.classList.remove(animationName));
+  ).then(() => el.classList.remove(animationName)).catch((error) => console.log(error));
```
nit: remove (unless you really wanna know when a target animation is cancelled. See comment below

KamalJDavis replied:
Fixed with the latest commit

streeeggs reviewed (commented):
LMK which comments were addressed and I can take a closer look

KamalJDavis committed:
- [x] initCounterForSelf now uses updateAnimation
- [x] Error logging from initCounterForSelf removed

KamalJDavis committed:
Updated Coolpoints user page.

KamalJDavis committed:
Merge branch '3.0' into rev-coolpoints-ui

streeeggs commented on www/css/coolhole.css:
```diff
@@ -400,16 +433,166 @@
+  display: flex;
+  justify-content: space-evenly;
+}
+.cpActionHeader:not([id="cpActionsCardHeader-Earnings"]) {
+  border-top: #ffffff50 1px solid;
+}
```
Is Earnings excluded because it's the first? Probably better to use [`:not:first-child`](https://developer.mozilla.org/en-US/docs/Web/CSS/:first-child) so we're not reliant on earnings always being first

KamalJDavis replied:
Damn, I must've missed that on MDN. I can use that instead.

streeeggs commented on www/css/coolhole.css:
```diff
@@ -400,16 +433,166 @@
+}
+.cpActionHeader {
+  padding: 10px 5px;
+  margin-top: 10px !important;
```
👀 What are we overriding here? Any way we can adjust or remove the rule causing this instead of relying on `!important`?

KamalJDavis replied:
I believe that something within bootstrap is marked `!important` that completely overtakes any custom margins here. This was my ham-fisted solution, I'll need to take a second look to find the best way to set this again.

KamalJDavis replied:
Finally checked it, the `h3` class from shale.css takes precedence over this class. To avoid using `!important` I can just make it `.cpActionHeader.h3` instead and it will function the same and be less intrusive.

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -24,9 +24,17 @@ function initPointsForSelf(pts) {
-  $("#coinWrapper").addClass("cpFadeIn");
 }
-
+/**
+ * Kamal - moved the button loading out of "initPointsForSelf()"" since it is called on page load.
+ * If we wanted to put any other animations on the button for points gained or earned we'd need the previous
+ * animation class removed i.e., "cpFadeIn".
+ * 9-13-24: Calls updateAnimation now as the previous code was redundant.
+ */
+function initCounterForSelf(){
+  const buttonEl = document.getElementById("cpPromptBtn");
+  updateAnimation(buttonEl.id, "cpFadeIn");
+}
```
See above. I think we should just render the button in always but render the counter if the user is valid. Currently, the `init` callback (and thus this function) won't fire for guests

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -72,9 +80,11 @@ function animatePointUpdate(ptEl, msgEl, diff) {
+  const glowAnimationName = isPositive ? "cpGlowGreen" : "cpGlowRed";
   const msgText = isPositive ? `+${diff}` : `${diff}`;
-
+  const btnEl = $("#cpPromptBtn");
```
nit: in the spirit of this function, mind passing this as an arg instead? I think i use this same function for updating the table when you update user's points. In that scenario we wouldn't have a button to update or a different button to update

KamalJDavis replied:
Yeah that's fine, I believe I was trying to make sure it was being targeted specifically to test the functionality.

streeeggs commented on www/css/coolhole.css:
```diff
@@ -17,19 +17,46 @@
+  animation-name: chGold2;
+  animation-duration: 30s;
+}
+@keyframes chGold2 {
```
nit: `2` seems like a shitty designation. is this just for a specific header in the prompt or any use of `code` with gold?

streeeggs commented on www/js/coolpoints-util.js:
```diff
@@ -24,9 +24,17 @@ function initPointsForSelf(pts) {
+ * Kamal - moved the button loading out of "initPointsForSelf()"" since it is called on page load.
+ * If we wanted to put any other animations on the button for points gained or earned we'd need the previous
+ * animation class removed i.e., "cpFadeIn".
+ * 9-13-24: Calls updateAnimation now as the previous code was redundant.
```
oh and, generally, no need to leave time stamped comments like this. git commit messages and history do a better job telling this "story" anyway

Commit:
modified:   www/css/coolhole.css
	modified:   www/js/coolpoints-util.js
	Removed coolpoints animation update function, rolled into main points update function.
	Renamed chGold2 to chGoldSlow.
	Removed "!important" in favor of targeting the classes directly.
	Comments removed from coolhole-util.js

esnyder232 reviewed (approved):
Looks good to me

streeeggs commented on www/css/coolhole.css:
```diff
@@ -3,718 +3,906 @@
+    display: flex;
+    justify-content: space-evenly;
+}
+.cpActionHeader.h3:not:first-child {
+    border-top: #ffffff50 1px solid;
+}
```
what's this style apply to?

streeeggs replied:
hey @streeeggs you dumbass, this one: https://github.com/streeeggs/coolhole/pull/21#discussion_r1786443264

streeeggs commented on www/css/coolhole.css:
```diff
@@ -3,718 +3,906 @@
+        background-position: 10em;
+    }
+}
+code.text-lottery {
```
note: might have the side effect of `monospaced` golds in chat. like those wrapped in "`". probably nbd

streeeggs reviewed (approved):
lgtm
